components/director/internal/domain/application/resolver.go

```diff
@@ -214,19 +214,21 @@ func (r *Resolver) SetApplicationLabel(ctx context.Context, applicationID string
 }
 
 func (r *Resolver) DeleteApplicationLabel(ctx context.Context, applicationID string, key string) (*graphql.Label, error) {
-	err := r.appSvc.DeleteLabel(ctx, applicationID, key)
+	app, err := r.appSvc.Get(ctx, applicationID)
```

Comment:
Do you really need to fetch the application? Underneath, there may be a lot of database calls, that are required to fetch all application object and you need here only the value of the label. In general, I'm not sure if we should return the actual value of the object that was deleted. For me, it is rather confusing.

Reply:
Unfortunately I need label values to return for GraphQL mutation result. But yes, I agree, we can think about these improvements later.